provider/kubernetes/kubernetes.go

```diff
@@ -308,6 +309,13 @@ func (p *Provider) loadIngresses(k8sClient Client) (*types.Configuration, error)
 				}
 			}
 		}
+
+		tlsConfigs, err := getTLSConfigurations(i, k8sClient)
+		if err != nil {
+			log.Errorf("Error configuring tls for ingress %s/%s: %s", i.Namespace, i.Name, err)
```

Comment:
I'd like to make sure we make a conscious decision on the implications for an Ingress that has an invalid TLS configuration or whose TLS data cannot be retrieved from the corresponding secret. IIUC, with the current approach we simply do not set the TLS configuration. Does that mean it'd not be accessible via HTTPS but possibly via HTTP? IMHO, that'd be something we shouldn't tolerate and instead skip that Ingress together with a clear error log message.

The line is also missing test coverage, but let's answer the question first.

Reply:
In my opinion, if an ingress is accessible via both http and https, and the https configuration fails, it should still be accessible via http.

However, if the TLS config fails on an https-only config, then yes, we should skip the ingress entirely.

Reply:
Is there a way to get the endpoints to which the configuration will be attached to?

Reply:
There are defaults, but the `traefik.frontend.entryPoints` annotation will override the defaults.

@timoreimann What do you think about adding another annotation to specify the TLS entrypoints to link to?
It feels super janky to me, since it seems to be like nailing a board with a screw. Ideally we would have access to the current entrypoint config to get the TLS-enabled entrypoints, and then use those, with the annotations overlaid if they exist.

It appears that the current k8s config does not access or reference the default entrypoints unless to override them with the annotation.